CRM/Core/BAO/CustomGroup.php

```diff
@@ -358,10 +358,17 @@ public static function &getTree(
     if ($entityID) {
       $entityID = CRM_Utils_Type::escape($entityID, 'Integer');
     }
+    if (!is_array($subTypes)) {
+      if (empty($subTypes)) {
+        $subTypes = array();
+      }
+      else {
+        $subTypes = explode(',', $subTypes);
```

Comment:
That commasep format is coming from uses like this -

* [`CRM_Contact_Form_Search_Custom_MultipleValues::__construct()`](https://github.com/civicrm/civicrm-core/blob/master/CRM/Contact/Form/Search/Custom/MultipleValues.php#L50)

That's the only place I found where I see a comma separated value passed in as $entityType, and I feel it ought to be an array so we can not have to handle this weird case. I haven't checked ALL the getTree() calls yet but have done most.

It probably can work, but it feels like something that should be handled by a standard SQL query generator instead of accepting a chunk of SQL from another function.